### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by jodh-intel**

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Found with your new linter tool per chance?
lgtm

File: src/process.c [hunk 1]
```diff
@@ -226,9 +226,7 @@ cc_oci_setup_shim (struct cc_oci_config *config,
 	ret = true;
 
 out:
-	if (tty_fd > 2) {
```
Comment (dlespiau) [inline src/process.c]: This seems to have been done on purpose in commit f1c21527e9056c3f629baaa98849c67edd866cbb

**Comment by chavafg** [comment]
qa-failed

File: src/process.c (hunk 1 above)
Comment (jamesodhunt) [inline src/process.c]: Indeed, but I don't think that is correct behaviour. I've just updated my test which should guarantee the fd gets closed in all scenarios.

@amshinde - could you review this?

Comment (grahamwhaley) [inline src/process.c]: /me admits it rang a little bell when he read it... iirc, this was so we do not close the host stdin/stdout/stderr iirc - and if so, I will nag that there was not enough comments in either the commit message nor the code... @amshinde - can you remember ? (and if so, maybe submit a new PR clarifying in the code etc. etc.)

Comment (dlespiau) [inline src/process.c]: I think the problem was: what happens if tty_fd is, say, 2? (and the fact that @amshinde had to introduce that case makes me think it's not a theoretical question).

- open(config->console, O_RDWR |  O_NOCTTY) returns 2
- dup2() on 0 and 1 work as usual, closing the previous 0 and 1
- dup2(tty_fd, 2) does nothing, from the man page:

```
    If  oldfd  is  a  valid  file descriptor, and newfd has the same value as oldfd, then dup2()
    does nothing, and returns newfd
```
close(tty_fd) will close fd 2, and we don't want to do that.

Comment (jodh-intel) [inline src/process.c]: Good catch! Test updated.

**Comment by dlespiau** [comment]
lgtm

File: src/process.c (hunk 1 above)
Comment (amshinde) [inline src/process.c]: lgtm. Yes tty_fd has been found to have values  <=2.

**Comment by amshinde** [comment]
lgtm

**Comment by jodh-intel** [comment]
Hi @chavafg - could you respin the tests for this PR?

**Comment by chavafg** [comment]
qa-passed

**Comment by devimc** [comment]
lgtm

**Pull request merged**